programs/zk-elgamal-proof/benches/verify_proofs.rs

```diff
@@ -0,0 +1,444 @@
+#![allow(clippy::arithmetic_side_effects)]
+use {
+    criterion::{criterion_group, criterion_main, Criterion},
+    curve25519_dalek::scalar::Scalar,
+    solana_zk_sdk::{
+        encryption::{
+            elgamal::ElGamalKeypair,
+            grouped_elgamal::GroupedElGamal,
+            pedersen::{Pedersen, PedersenOpening},
+        },
+        zk_elgamal_proof_program::proof_data::{
+            BatchedGroupedCiphertext2HandlesValidityProofData,
+            BatchedGroupedCiphertext3HandlesValidityProofData, BatchedRangeProofU128Data,
+            BatchedRangeProofU256Data, BatchedRangeProofU64Data,
+            CiphertextCiphertextEqualityProofData, CiphertextCommitmentEqualityProofData,
+            GroupedCiphertext2HandlesValidityProofData, GroupedCiphertext3HandlesValidityProofData,
+            PercentageWithCapProofData, PubkeyValidityProofData, ZeroCiphertextProofData,
+            ZkProofData,
+        },
+    },
+};
+
+fn bench_pubkey_validity(c: &mut Criterion) {
+    let keypair = ElGamalKeypair::new_rand();
+    let proof_data = PubkeyValidityProofData::new(&keypair).unwrap();
+
+    c.bench_function("pubkey_validity", |b| {
+        b.iter(|| {
+            proof_data.verify_proof().unwrap();
+        })
+    });
+}
+
+fn bench_zero_ciphertext(c: &mut Criterion) {
+    let keypair = ElGamalKeypair::new_rand();
+    let ciphertext = keypair.pubkey().encrypt(0_u64);
+    let proof_data = ZeroCiphertextProofData::new(&keypair, &ciphertext).unwrap();
+
+    c.bench_function("zero_ciphertext", |b| {
+        b.iter(|| {
+            proof_data.verify_proof().unwrap();
+        })
+    });
+}
+
+fn bench_grouped_ciphertext_2_handles_validity(c: &mut Criterion) {
+    let destination_keypair = ElGamalKeypair::new_rand();
+    let destination_pubkey = destination_keypair.pubkey();
+
+    let auditor_keypair = ElGamalKeypair::new_rand();
+    let auditor_pubkey = auditor_keypair.pubkey();
+
+    let amount: u64 = 55;
```

Comment:
This is just for my own knowledge, I can't remember if I've asked this before, so apologies if I have.

Is there any difference in execution time between large and small numbers?

Reply:
Nope, the amount is converted into a (randomized) scalar and scalar arithmetic is implemented in constant time, so all numbers should have comparable performance.

Reply:
ok great, thanks!